lib/components/CursorTooltip/CursorTooltip.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useRef } from 'react'
+import { Tooltip } from '@mui/material'
+import clsx from 'clsx'
+import { EMPTY_STRING } from '../../consts'
+
+import '../Tooltip/tooltip.scss'
+
+export type CursorTooltipProps = {
+  children: React.ReactElement
+  clear?: boolean
+  data?: React.ReactElement | string
+  extraClass?: string
+  extraPopperClass?: string
+  showDelay?: number
+  disappearAfterTimeout?: number
+} & React.ComponentProps<typeof Tooltip>
+
+function CursorTooltip({
+  children,
+  clear = false,
+  data = EMPTY_STRING,
+  extraClass = EMPTY_STRING,
+  extraPopperClass = EMPTY_STRING,
+  showDelay = 1000,
+  disappearAfterTimeout = Infinity,
+  ...rest
+}: CursorTooltipProps) {
+  const [open, setOpen] = useState(false)
+  const showTimeoutRef = useRef<NodeJS.Timeout | null>(null)
+  const disappearTimeoutRef = useRef<NodeJS.Timeout | null>(null)
+  const isInsideContainer = useRef(false)
+
+  const handleMouseEnter = () => {
+    isInsideContainer.current = true
+    if (disappearTimeoutRef.current) {
+      clearTimeout(disappearTimeoutRef.current)
+    }
+    showTimeoutRef.current = setTimeout(() => {
+      if (isInsideContainer.current) {
+        setOpen(true)
+        if (disappearAfterTimeout !== Infinity) {
+          disappearTimeoutRef.current = setTimeout(() => {
+            setOpen(false)
+          }, disappearAfterTimeout)
+        }
+      }
+    }, showDelay)
+  }
+
+  const handleMouseLeave = () => {
+    isInsideContainer.current = false
+    if (showTimeoutRef.current) {
+      clearTimeout(showTimeoutRef.current)
+    }
+    if (disappearTimeoutRef.current) {
+      clearTimeout(disappearTimeoutRef.current)
+    }
+    setOpen(false)
+  }
+
+  const handleMouseMove = () => {
+    if (open) {
+      setOpen(false)
+      handleMouseEnter()
+    }
+  }
+
+  const classes = clsx({
+    'tooltip-wrapper': true,
+    'clear-tooltip': clear,
+    [extraClass]: true
+  })
+
+  return (
+    <Tooltip
+      {...rest}
+      open={open}
+      title={data}
+      classes={{
+        tooltip: classes,
+        arrow: 'tooltip-arrow',
+        popper: extraPopperClass
+      }}
+      placement='top'
```

Comment:
maybe let's leave the placement to be dynamic as well?

Reply:
It is the same behavior also in the regular `tooltip` we can override it since it will be merged with the "rest" prop (I'll move it down)